src/search/utils/exceptions.h

```diff
@@ -1,12 +1,19 @@
 #ifndef UTILS_EXCEPTIONS_H
 #define UTILS_EXCEPTIONS_H
 
+#include <string>
+
 namespace utils {
 // Base class for custom exception types.
 class Exception {
+protected:
+    const std::string msg;
 public:
+    explicit Exception(const std::string &msg);
     virtual ~Exception() = default;
-    virtual void print() const = 0;
+
+    std::string get_message() const;
+    virtual void print() const;
```

Comment:
Since Exception now has a print() method, do we need to add override for derived classes that define their own print method? (E.g. ArgError in command_line)

Reply:
Yes, but this is done in all three derived classes. Did we miss it somewhere?

Reply:
Oups, my notes were wrong, they said print() is a new method and thus I was thinking we need to add override and did not remember seeing that change anywhere. But since the print method existed already the overrides are of course already there.